evap/results/exporters.py

```diff
@@ -122,53 +135,58 @@ def filter_evaluations(semesters, evaluation_states, degrees, course_types, cont
                 continue
             if not evaluation.can_publish_rating_results and not include_not_enough_voters:
                 continue
-            results = OrderedDict()
+            results: OrderedDict[EvaluationResult, list[QuestionResult]] = OrderedDict()
             for contribution_result in get_results(evaluation).contribution_results:
                 for questionnaire_result in contribution_result.questionnaire_results:
                     # RatingQuestion.counts is a tuple of integers or None, if this tuple is all zero, we want to exclude it
+                    question_results: list[QuestionResult] = questionnaire_result.question_results  # type: ignore[assignment]
                     if all(
                         not question_result.question.is_rating_question
-                        or question_result.counts is None
-                        or sum(question_result.counts) == 0
-                        for question_result in questionnaire_result.question_results
+                        or (
+                            isinstance(question_result, RatingResult)
+                            and (question_result.counts is None or sum(question_result.counts) == 0)
+                        )
+                        for question_result in question_results
```

Comment:
When we were talking about this, I had something like this in my mind:
```python3
if all(
    not isinstance(question_result, RatingResult)
    or question_results.counts is None
    or sum (question_result.counts) == 0
    for question_result in question_results
):
```

Reply:
(Note: `isinstance(question_result, RatingResult)` is 'wider' than the old `question_result.question.is_rating_question` since we can now have text answers to rating questions. I believe that my suggestion here is what the intention behind this code was, but this might be wrong)

Reply:
Not happy with the current change here since it modifies behavior. Before, if we had a text answer to a rating question, this code crashed since `question_result.counts` undefined. Now, it would simply pass and include that result.

From a quick look at `_get_results_impl`, it looks to me as if additional text answers to rating questions are always included in a rating result. If those never existed as a TextAnswer, then the "text answer to rating question" situation shouldn't be a problem here. This would explain why we never observed a crash here.

I currently believe that the `not question_result.question.is_rating_question` part was initially intended to filter out all non-rating-results and is from before we had the `RatingResult` class, so that's why it doesn't use `isinstance`.